framework/DataObjects/XDataSet.py

```diff
@@ -206,13 +206,52 @@ def addVariable(self,varName,values,classify='meta'):
       self._metavars.append(varName)
     self._allvars.append(varName)
 
-  def asDataset(self):
+  def asDataset(self, outType=None):
+    """
+      Casts this dataObject as dictionary or an xr.Dataset depending on outType.
+      @ In, outType, str, type of output object (xr.Dataset or dictionary).
+      @ Out, xr.Dataset or dictionary.
+    """
+    if outType=='None' or outType=='xrDataset':
+      # return the old asDataset()
+      return self._convertToXrDataset()
+    elif outType=='dict':
+      # return a dict
+      return self._convertToDict()
+    else:
+      # raise an error
+      print('raise an error')
+
+  def _convertToDict(self):
+    """
+      Casts this dataObject as dictionary.
+      @ In, None
+      @ Out, data, dict, dictionary containing for all.
+    """
+    dataDict={}
+
+    dataDict['data'] = {}
+    for var in self.getVars():
+      dataDict['data'][var] = self.asDataset()[var]
+
+    for var in self.indexes:
+      length = self.sliceByIndex(var).size
+      dataDict['data'][var] = np.zeros(length,dtype=object)
+      for elem in self.sliceByIndex(var):
+        dataDict['data'][var] = elem
+
+    dataDict['dimensions'] = self.getDimensions('output')
+    dataDict['metadata']   = self.getMeta(pointwise=True, general=True)
```

Comment:
Don't do `pointwise=True` here, because you already included pointwise in `for var in self.getVars()`